File: #04/q12 - rotDir/rotDir.hs


rotDir :: (Eq t, Num t) => t -> [a] -> [a]
rotDir a [] = [] 
rotDir 0 xs = xs
rotDir a xs = rotDir (a-1) (last xs : init xs)


main :: IO ()
main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ rotDir a b

File: #04/q12 - rotDir/rotDir.js
rotDir = (x, arr) => {
    if (arr.length == 0) return []
    else if (x == 0) return arr
    else {
        arr.push(arr.shift())
        return rotDir(x-1, arr)
    }
}

console.log(rotDir(0, [1,2,3]))
console.log(rotDir(1, [1,2,3]))
console.log(rotDir(2, [1,2,3]))